### DISCUSSION TITLE
Different Database Entegration

### DISCUSSION THREAD

**Discussion by drvshavva** [apache/streampipes#2460] [Q&A] [answered]
What do I need to do for a different database integration? Since InfluxDB is not an open-source database, its use will limit us. In this case, if I wanted to use a new database, how could I position it in the architecture? And what is your reason for choosing InfluxDB?

**Comment by tenthe** [comment]
Hi @drvshavva,

currently only InfluxDB is supported. The community also discussed switching to IoTDB, see #1369.
There have been initial efforts to integrate this, but this is not yet finalized.
There is also an initial PR in the discussion to replace InfluxDB with IoTDB. 

If this integration is exciting to you, we can discuss how to integrate IoTDB. 
It would be particularly interesting to compare the differences in terms of performance and functionality.

Cheers,
Philipp

**Reply by drvshavva** [reply]
Thank you for your prompt response, I am also excitedly following the work there :)

**Comment by dominikriemer** [comment]
Hi  @drvshavva just one addition since you're saying InfluxDB isn't an open source database, the Influx version we use (v2.x) is fully open source and MIT-licensed.
We however still plan to replace the database but there is not a timeframe yet.

**Reply by drvshavva** [reply, marked as answer]
Hello, thank you very much for your answers and for the information, I didn't know that version was open source. :)
